kubespawner/reflector.py

```diff
@@ -172,26 +172,29 @@ def __init__(self, *args, **kwargs):
         self.first_load_future = Future()
         self._stop_event = threading.Event()
 
-        # Make sure that we know kind, whether we should omit the namespace,
-        #  and what our list_method_name is.  For the things we already
-        #  know about (that is, Pod and Event reflectors) we can derive
-        #  list_method_name from those two things.  New reflector types
-        #  should also update their __init__() methods to derive
-        #  list_method_name, but you could just set it directly in the
-        #  subclass.
+        # Make sure that we know kind, whether we should omit the
+        #  namespace, and what our list_method_name is.  For the things
+        #  we already know about, we can derive list_method_name from
+        #  those two things.  New reflector types should also update
+        #  their __init__() methods to derive list_method_name, but you
+        #  could just set it directly in the subclass.
         if not self.list_method_name:
             # This logic can be extended if we add other reflector types or
             #  it can be directly supplied or overridden in a subclass.
-            if self.kind == "pods":
-                if self.omit_namespace:
-                    self.list_method_name = "list_pod_for_all_namespaces"
+            if self.kind in ["pods", "events", "services", "ingresses", "endpoints"]:
+                if self.kind == "ingresses":
+                    singular_kind = "ingress"
+                if self.kind == "endpoints":
+                    # This handles inconsistent nomenclature in the
+                    # Python Kubernetes client.
+                    singular_kind = self.kind
                 else:
-                    self.list_method_name = "list_namespaced_pod"
-            elif self.kind == "events":
+                    singular_kind = self.kind[:-1]
```

Comment:
Can we instead just list singular and plurals in a dictionary instead? Automatically trying to figure out singulars from plurals usually gets filled with exceptions anyway, and I think that's clearer.

Reply:
You're 1000x right. 

Pushed up the pivot to a dictionary. 

Validated tests are passing and execute pre-commit.